components/org.wso2.carbon.identity.organization.management.service/src/main/java/org/wso2/carbon/identity/organization/management/service/OrganizationManagerImpl.java

```diff
@@ -434,17 +435,20 @@ private List<Organization> getOrganizationList(Integer limit, String after, Stri
     @Override
     public String getOrganizationVersion(String organizationId) throws OrganizationManagementException {
 
-        if (StringUtils.isEmpty(organizationId)) {
-            throw handleClientException(ERROR_CODE_ORGANIZATION_ID_UNDEFINED);
+        String tenantDomain = resolveTenantDomain(organizationId);
+        if (OrganizationManagementUtil.isOrganization(tenantDomain)) {
+            // If the given organization ID is of a sub-organization, fetch the version of the primary organization.
+            organizationId = getPrimaryOrganizationId(organizationId);
+            tenantDomain = resolveTenantDomain(organizationId);
         }
 
-        Organization organization = organizationManagementDAO.getOrganization(organizationId);
-        if (organization == null) {
+        Optional<String> organizationVersion = organizationManagementDAO.getOrganizationVersion(organizationId,
+                tenantDomain);
+        if (!organizationVersion.isPresent()) {
             throw handleClientException(ERROR_CODE_INVALID_ORGANIZATION, organizationId);
```

Comment:
In sub org's version resolving case cause for this situation, the root org's id will be passed in this client exception.

Reply:
organizationVersion will be null only if no entry exists in the UM_ORG table for the given organization ID. In such cases, the tenant domain will also be null, causing the isOrganization check to return false, and the sub-organization’s ID will be used instead.